ledger/internal/assetcow.go

```diff
@@ -23,22 +23,28 @@ import (
 
 func (cs *roundCowState) AllocateAsset(addr basics.Address, index basics.AssetIndex, global bool) error {
 	if global {
-		cs.mods.Creatables[basics.CreatableIndex(index)] = ledgercore.ModifiedCreatable{
-			Ctype:   basics.AssetCreatable,
-			Creator: addr,
-			Created: true,
-		}
+		cs.mods.UpsertCreatable(
+			basics.CreatableIndex(index),
+			ledgercore.ModifiedCreatable{
+				Ctype:   basics.AssetCreatable,
+				Creator: addr,
+				Created: true,
+			},
+		)
 	}
 	return nil
 }
 
 func (cs *roundCowState) DeallocateAsset(addr basics.Address, index basics.AssetIndex, global bool) error {
 	if global {
-		cs.mods.Creatables[basics.CreatableIndex(index)] = ledgercore.ModifiedCreatable{
-			Ctype:   basics.AssetCreatable,
-			Creator: addr,
-			Created: false,
-		}
+		cs.mods.UpsertCreatable(
+			basics.CreatableIndex(index),
+			ledgercore.ModifiedCreatable{
+				Ctype:   basics.AssetCreatable,
+				Creator: addr,
+				Created: false,
+			},
+		)
```

Comment:
could write a quick test and see if the new code has more allocations than the old one? - wondering if compiler optimizes it right.

Reply:
Did you mean for this function in particular? or in general?

I ran the `BenchmarkMakeStateDelta` that already existed. Out of the box it's the same on master and this branch at 5allocs/op but when adding `//go:noinline` to `MakeStateDelta` the difference becomes 7 vs 12 as expected. 

That seems odd to me since running `go build --gcflags="-m" .` in `ledger/ledgercore` gives this showing that the allocs are escaping to heap when compiled and scenario1 run above confirms it:
<img width="679" alt="Screen Shot 2022-11-03 at 3 27 01 PM" src="https://user-images.githubusercontent.com/1126091/199817561-d4d256fe-6a8f-4b25-9f67-fc0d00eb98d3.png">

Reply:
then maybe `AddCreatable` should take a pointer to get rid of additional copying?

Reply:
It looks like this is already merged but do you want me to add it? This should be called rarely and the objects are tiny so the copy cost is not big but should be safe